Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -14054,6 +14089,354 @@ def pan_os_get_audit_comment_command(args: dict) -> CommandResults:
+            "Profile name was found both in Vulnerability Protection Profiles and in Anti Spyware Profiles. Please specify profile_type.")
+
+    elif profile_name in vulnerability_protection_profile_names:
+        return VULNERABILITY_PROTECTION
+
+    elif profile_name in anti_spyware_profile_names:
+        return ANTI_SPYWARE
+
+    else:
+        raise DemistoException("Profile name was not found in Vulnerability Protection Profiles or in Anti Spyware Profiles.")
+
+
+def build_xpath_for_profile_exception_commands(profile_name: str, profile_type: str, device_group: str, action_type: str, extracted_id: Optional[str] = None) -> str:
+    """
+    Creates and return xpath based on the profile type and pan-os/panorama instance.
+
+    Args:
+        profile_name: The profile name.
+        profile_type: The profile type.
+        device_group: The device group if was sent as a commands argument.
+        action_type: Action type for api request.
+        extracted_id: The id of the exception.
+
+    Returns:
+        The xpath.
+    """
+
+    if profile_type == VULNERABILITY_PROTECTION and device_group:
+        xpath = VULNERABILITY_PROTECTION_DEVICE_GROUP_PATH.format(device_group=device_group, profile_name=profile_name)
+
+    elif profile_type == VULNERABILITY_PROTECTION and VSYS:
+        xpath = VULNERABILITY_PROTECTION_VSYS_PATH.format(VSYS=VSYS, profile_name=profile_name)
+
+    elif profile_type == ANTI_SPYWARE and device_group:
+        xpath = ANTI_SPYWARE_DEVICE_GROUP_PATH.format(device_group=device_group, profile_name=profile_name)
+
+    elif profile_type == ANTI_SPYWARE and VSYS:
+        xpath = ANTI_SPYWARE_VSYS_PATH.format(VSYS=VSYS, profile_name=profile_name)
+
+    else:
+        raise DemistoException("Invalid profile_type was provided. Can be Vulnerability Protection or Anti Spyware.")
+
+    if action_type in [EDIT_EXCEPTION_COMMAND_TYPE, DELETE_EXCEPTION_COMMAND_TYPE]:
+        xpath += f"/entry[@name='{extracted_id}']"
+
+    return xpath
+
+
+def get_predefined_threats_list() -> list:
+    """
+    Get the predefined threats lists.
+
+    Returns:
+        The list of threats.
+    """
+    result = panorama_get_predefined_threats_list()
+    predefined_threats = result['response']['result']['threats']["phone-home"]['entry']
+    predefined_threats += result['response']['result']['threats']["vulnerability"]['entry']
+    return predefined_threats
+
+
+def get_threat_id_from_predefined_threates(threat_name: str) -> str:
+    """
+    Search the threat id in the threats list by using the threat_name argument.
+
+    Args:
+        threat_name: The threat name, can be the CVE or the id.
+
+    Returns:
+        The id that fit to the threat_name
+    """
+    predefined_threats = get_predefined_threats_list()
+
+    for entry in predefined_threats:
+        search_keys = []
+
+        search_keys.append(entry.get("threatname", ""))
+
+        extracted_id = entry["@name"]
+        search_keys.append(extracted_id)
```

Comment:
extracted_id is the human readable text?

Reply:
extracted_id is the id of the exception